homeassistant/components/image_processing/clarifai_general.py

```diff
@@ -0,0 +1,141 @@
+"""
+Component that will perform image classification via Clarifai general model.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/image_processing/clarifai_general
+"""
+import base64
+import json
+import logging
+
+import voluptuous as vol
+
+from homeassistant.core import split_entity_id
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.image_processing import (
+    PLATFORM_SCHEMA, ImageProcessingEntity, CONF_SOURCE, CONF_ENTITY_ID,
+    CONF_NAME)
+from homeassistant.const import STATE_UNKNOWN
+
+
+_LOGGER = logging.getLogger(__name__)
+
+CLASSIFIER = 'Clarifai'
+CONF_API_KEY = 'api_key'
+
+REQUIREMENTS = ['clarifai==2.4.1']
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_API_KEY): cv.string,
+})
+
+
+def encode_image(image):
+    """base64 encode an image stream."""
+    base64_img = base64.b64encode(image)
+    return base64_img
+
+
+def parse_concepts(api_concepts):
+    """Parse the API concepts data."""
+    return {concept['name']: round(100.0*concept['value'], 2)
+            for concept in api_concepts}
+
+
+def validate_api_key(api_key):
+    """Check that an API key is valid, if yes return the app."""
+    try:
+        from clarifai.rest import ClarifaiApp, ApiError
+        app = ClarifaiApp(api_key=api_key)
+        return app
+    except ApiError as exc:
+        error = json.loads(exc.response.content)
+        _LOGGER.error(
+            "%s error: %s", CLASSIFIER, error['status']['description'])
```

Comment:
You can get more information about the error by simply outputting `str(exc)` - HTTP code, status code, description, erorr details, REST url etc. are displayed.

Note: In the version 2.4.1 there's some bug in the output of `str(exc)` where not all details are displayed correctly - but this will get fixed in a few days when 2.4.2 is released.

Reply:
I've experimented with this a bit, but think we can stick with the `description` for now